lol/esportsapi/esports-match-data.md

```diff
@@ -0,0 +1,230 @@
+    Returns the list server, gameId and hashes of games played in a tournament.
+
+    :param tournament_name
+                Name of the tournament, which can be gotten from get_tournaments().
+    :return:
+                A list of game dictionaries.
+    """
+    games = self._cargoquery(tables='ScoreboardGames',
+                             fields='Tournament = tournament, '
+                                    'MatchHistory = match_history_url, ',
+                             where="ScoreboardGames.Tournament='{}'".format(tournament_name),
+                             order_by="ScoreboardGames.DateTime_UTC",
+                             **kwargs)
+    data = [
+        {
+            "tournament":game["tournament"],
+            "server":game["match_history_url"].split("/")[5],
+            "gameId":game["match_history_url"].split("/")[6].split("?gameHash=")[0],
+            "hash":game["match_history_url"].split("/")[6].split("?gameHash=")[1],
+        }
+        for game in games
+    ]
+    return data
+
+lp.get_games_hashes = types.MethodType(get_games_hashes, lp)
+```
+
+## Getting the hashes
+
+Getting the hashes for LEC 2020 Summer : 
+
+
+```python
+games = lp.get_games_hashes(tournaments[3]['name'])
+games[:3]
+```
+
+
+
+
+    [{'tournament': 'LEC 2020 Summer',
+      'server': 'ESPORTSTMNT04',
+      'gameId': '1230688',
+      'hash': '25cb7e1966cbcdb5'},
+     {'tournament': 'LEC 2020 Summer',
+      'server': 'ESPORTSTMNT04',
+      'gameId': '1220706',
+      'hash': 'c3f45e5bb2a65c80'},
+     {'tournament': 'LEC 2020 Summer',
+      'server': 'ESPORTSTMNT04',
+      'gameId': '1220728',
+      'hash': '4bfd5c00f9292be3'}]
+
+
+
+Requesting the match data from all those games : 
+
+
+```python
+import requests
+
+base_match_history_stats_url = "https://acs.leagueoflegends.com/v1/stats/game/{}/{}?gameHash={}"
+base_match_history_stats_timeline_url = "https://acs.leagueoflegends.com/v1/stats/game/{}/{}/timeline?gameHash={}"
+
+all_games_data = []
+
+for g in games:
+    url = base_match_history_stats_url.format(g["server"],g["gameId"],g["hash"])
+    timeline_url = base_match_history_stats_timeline_url.format(g["server"],g["gameId"],g["hash"])
+    
+    game_data = requests.get(url).json()
+    game_data["timeline"] = requests.get(timeline_url).json()
+    
+    all_games_data.append(game_data)
+```
+
+## In need of Cookies
+
+ACS is highly rate limited and you might need to authenticate yourself to avoid this.
+
+For that, you will need to : 
+ * Open your browser
+ * Log in League of Legends website (or any one dependent on Riot log in)
+ * Open an acs link (e.g. https://acs.leagueoflegends.com/v1/stats/game/ESPORTSTMNT06/1070986?gameHash=1787d1ea36a382c3 )
+ * Right clic -> Inspect element
+ * Network tab
+ * Select the first item in the network list (might need a refresh with the tab open)
+ * Find the request header
+ * Copy the value of the Cookie header
+
+![ACS header](http://canisback.com/images/acs.png)
```

Comment:
Since we currently don't proxy the images on our end, we're gonna need an HTTPS URL